ietf/context_processors.py

```diff
@@ -51,14 +39,11 @@ def social_menu(site):
 
 def global_pages(request):
     site = Site.find_for_request(request)
+    # XXX Return lazy values. This makes a big difference when a page renders
+    # multiple templates, e.g. when the wagtail userbar is displayed.
     return {
-        "HOME": home_page(site),
-        "BLOG_INDEX": BlogIndexPage.objects.first(),
-        "MENU": menu(site),
-        "SECONDARY_MENU": secondary_menu(site),
-        "SOCIAL_MENU": social_menu(site),
-        "BASE_URL": getattr(settings, "WAGTAILADMIN_BASE_URL", ""),
-        "DEBUG": getattr(settings, "DEBUG", ""),
-        "FB_APP_ID": getattr(settings, "FB_APP_ID", ""),
-        "SITE": site,
+        "HOME": lambda: home_page(site),
+        "MENU": lambda: get_main_menu(site),
+        "SECONDARY_MENU": lambda: secondary_menu(site),
+        "SOCIAL_MENU": lambda: social_menu(site),
```

Comment:
Doesn't this need to return `BLOG_INDEX` anymore?

Reply:
No, it was removed from templates in 53c3e66, and not referenced anywhere in the code now.